src/main/kotlin/net/lumalyte/lg/interaction/commands/GuildCommand.kt

```
@@ -2069,70 +2069,44 @@ class GuildCommand : BaseCommand(), KoinComponent {
-            return
-        }
-
-        // Check if ItemShops plugin is available
-        val itemShopsPlugin = Bukkit.getPluginManager().getPlugin("ItemShops")
-        if (itemShopsPlugin == null || !itemShopsPlugin.isEnabled) {
-            player.sendMessage("§c❌ ItemShops plugin not found!")
+        // Register the shop at that sign as guild-owned. Guard the cross-plugin call so a
+        // thrown exception gives a deterministic failure instead of an uncontrolled command error.
+        val converted = try {
+            shopLookup.registerGuildShopBySign(targetSign.location, guild.id, playerId)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)